### DISCUSSION TITLE
Is it possible to have AppDataRef's lifetime non-static?

### DISCUSSION THREAD

**Discussion by HughHoyland** [mlua-rs/mlua#376] [Q&A] [answered]
After migrating from an earlier `rlua`,  I discovered this amazing feature - that I can store data, even references (!!!) in `Lua` object.

But `set_app_data` only takes values with `'static` lifetimes.

I hoped I can store a reference that outlives Lua object itself. Is it, in theory, possible? I could create a fork where `Lua` has a lifetime parameter, `Lua<'lua>` - any reasons it's a bad idea?

**Comment by khvzak** [comment, marked as answer]
Only `'static` types has [`TypeId`](https://doc.rust-lang.org/stable/std/any/struct.TypeId.html). The easiest option would be casting type to static through [transmute](https://doc.rust-lang.org/stable/std/mem/fn.transmute.html) (unsafe)
